### PR TITLE
Support github app for api access

### PR DESCRIPTION
**Pull request by hallipr**

### CODE REVIEW

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.Tokens;
+using Octokit;
+
+namespace Azure.Sdk.Tools.PipelineWitness.GitHubActions;
+
+public class GitHubClientFactory
+{
+    private readonly TimeSpan processTimeout = TimeSpan.FromSeconds(13);
```
Comment (weshaggard) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs:20]: Why 13 seconds?

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        };
+
+        using ProcessRunner processRunner = new(process, this.processTimeout, CancellationToken.None);
+
+        string output = await processRunner.RunAsync().ConfigureAwait(false);
+
+        return new Credentials(output, AuthenticationType.Bearer);
+    }
+
+    private static ProcessStartInfo GetAzureCliProcessStartInfo()
```
Comment (weshaggard) [inline tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs:142]:
```suggestion
    private static ProcessStartInfo GetGitHubCliProcessStartInfo()
```

**Review by weshaggard — APPROVED** [review]

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs (hunk 1 above)
Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs:20]: That's a great question.   Much of the CLI interaction was lifted from [AzureCliCredential](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/src/Credentials/AzureCliCredential.cs#L80) and their default is 13 seconds

File: tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs (hunk 2 above)
Comment (hallipr) [reply tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/GitHubActions/GitHubClientFactory.cs]: Thanks.  Changed

**Comment by hallipr** [comment]
/check-enforcer override

There is no PR pipeline for pipeline-witness

**Pull request merged**